### PR TITLE
Vscode 71

### PR DESCRIPTION
**Pull request by mbarekh**
Add a XComponent transition snippet

### CODE REVIEW

File: extension/snippets/xcomponent.json [hunk 1]
```diff
@@ -2,10 +2,19 @@
 	"Create a XComponent state": {
 		"prefix": "xcstate",
```
Comment (maurelio1234) [inline extension/snippets/xcomponent.json]: could we stop putting xc everywhere?

@nicolaserny

File: extension/snippets/xcomponent.json [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
+			"<StateData Id=\"${1:index}\" Name=\"${2:state name}\" IsEntryPoint=\"${3|false,true|}\" SubGraphKey=\"${4:StateMachine + Id}\">",
+			"\t<TriggeredMethods />",
+			"</StateData>"
+		]
+	},
+	"Create a XComponent transition": {
+		"prefix": "xctransition",
+		"description": "Create a XComponent transition",
+		"body": [
+			"<TransitionData Id=\"${1:index}\" Name=\"${2:transition name}\" FromKey=\"${3:State + Id}\" ToKey=\"${4:State + Id}\" Type=\"${5:X}\" ExecutionDelay=\"${6:0}\" SetCustomTimeout=\"${7|false,true|}\" TriggeringEvent=\"${8:TriggeringEvent}\" UserSpecificRule=\"${9|false,true|}\">",
```
Comment (maurelio1234) [inline extension/snippets/xcomponent.json]: There's only one type? X?

File: extension/snippets/xcomponent.json [hunk 3]
```diff
@@ -2,10 +2,19 @@
 	"Create a XComponent state": {
 		"prefix": "xcstate",
 		"description": "Create a XComponent state",
-    	"body": [
-            "<StateData Id=\"${1:index}\" Name=\"${2:state name}\" IsEntryPoint=\"${3|false,true|}\" SubGraphKey=\"${4:StateMachine + Id}\">",
-            "\t<TriggeredMethods />",
-            "</StateData>"
+		"body": [
+			"<StateData Id=\"${1:index}\" Name=\"${2:state name}\" IsEntryPoint=\"${3|false,true|}\" SubGraphKey=\"${4:StateMachine + Id}\">",
```
Comment (maurelio1234) [inline extension/snippets/xcomponent.json]: Index 
State name

File: extension/snippets/xcomponent.json (hunk 2 above)
Comment (maurelio1234) [reply extension/snippets/xcomponent.json]: "Transition name"
Document Triggering event

Comment (maurelio1234) [reply extension/snippets/xcomponent.json]: can we add more information on the description?

File: extension/snippets/xcomponent.json (hunk 1 above)
Comment (nicolaserny) [reply extension/snippets/xcomponent.json]: I think it is a good idea for snippets and a common practice.
You can easily find the xcomponent snippets with intellisense by typing xc...

Comment (maurelio1234) [reply extension/snippets/xcomponent.json]: Ok :)

File: extension/snippets/xcomponent.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
 		"prefix": "xctransition",
 		"description": "Create a XComponent transition",
 		"body": [
-			"<TransitionData Id=\"${1:index}\" Name=\"${2:transition name}\" FromKey=\"${3:State + Id}\" ToKey=\"${4:State + Id}\" Type=\"${5:X}\" ExecutionDelay=\"${6:0}\" SetCustomTimeout=\"${7|false,true|}\" TriggeringEvent=\"${8:TriggeringEvent}\" UserSpecificRule=\"${9|false,true|}\">",
+			"<TransitionData Id=\"${1:Index}\" Name=\"${2:Transition name}\" FromKey=\"${3:State + Id}\" ToKey=\"${4:State + Id}\" Type=\"${5:|X,TimeOut,Standard|}\" ExecutionDelay=\"${6:0}\" SetCustomTimeout=\"${7|false,true|}\" TriggeringEvent=\"${8:Document Triggering event}\" UserSpecificRule=\"${9|false,true|}\">",
```
Comment (nicolaserny) [inline extension/snippets/xcomponent.json]: Wrong definition of the type enum
it should be ${5|X,TimeOut,Standard|} instead

**Pull request merged**